### PR TITLE
Java to Kotlin Nullability guide

### PR DESCRIPTION
**Pull request by PetrakovichVictoria**

### CODE REVIEW

File: docs/topics/jvm/java-to-kotlin-nullability-guide.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+> In the Java example above, the function `getStringLength()` returns a result of the primitive type `int`.
+To make it return `null`, you can use the ["boxed" type](https://docs.oracle.com/javase/tutorial/java/data/autoboxing.html) `Integer`.
+However, it's more resource-efficient to make such functions return a negative value and then check the value –
+you would do the check anyway, but no additional boxing is performed this way.
+>
+> {type="note"}
+
+## What's next?
+
+* Visit [Kotlin Koans](koans.md) – complete exercises to learn Kotlin syntax. Each exercise is created as a failing unit test,
```
Comment (koshachy) [inline docs/topics/jvm/java-to-kotlin-nullability-guide.md:320]: Too many links here.
At least we could remove Koand links: I've checked the analytics for the Collections guide for the last three months and found that there are a few of the clicks (33). What do you think?

**Review by koshachy — CHANGES_REQUESTED** [review]

File: docs/topics/jvm/java-to-kotlin-nullability-guide.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+System.out.println(numbers.get(0));
+//numbers.get(5) // Exception!
+```
+{id="functions-returning-null-java"}
+
+The Kotlin standard library often provides functions whose names indicate whether they can possibly return a `null` value.
+This is especially common in the collections API:
+
+```kotlin
+// Kotlin
```
[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/jvm/java-to-kotlin-nullability-guide.md:235]: I think we should move it to be visible. I think in such guides we should mark every code block. I probably missed it somewhere; I'll check.

[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/jvm/java-to-kotlin-nullability-guide.md:235]: >Also, some codeblocks in Kotlin don't contain such comments, but some single line code blocks have them.

Talking about this guide or overall? I checked the guide; this and one more place are the only where this comment was absent in the right place.

Which. single line code blocks are you talking about?

File: docs/topics/jvm/java-to-kotlin-nullability-guide.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+System.out.println("Max: " + max);
+```
+{id="aggregate-functions-java"}
+
+In Kotlin, use [aggregate operations](collection-aggregate.md):
+
+```kotlin
+// Kotlin
+fun main() {
+//sampleStart
```
[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/jvm/java-to-kotlin-nullability-guide.md:270]: No, it shouldn't. I'll correct it.

File: docs/topics/jvm/java-to-kotlin-nullability-guide.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+
+fun getStringLength(y: Any): Int {
+    val x: String? = y as? String // null
+    return x?.length ?: -1 // Returns -1 because `x` is null
+}
+```
+{kotlin-runnable="true" id="casting-types-kotlin"}
+
+> In the Java example above, the function `getStringLength()` returns a result of the primitive type `int`.
+To make it return `null`, you can use the ["boxed" type](https://docs.oracle.com/javase/tutorial/java/data/autoboxing.html) `Integer`.
```
[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/jvm/java-to-kotlin-nullability-guide.md:312]: Yes, would be better.

File: docs/topics/jvm/java-to-kotlin-nullability-guide.md (hunk 1 above)
Comment (PetrakovichVictoria) [reply docs/topics/jvm/java-to-kotlin-nullability-guide.md:320]: I agree.

File: docs/topics/null-safety.md [hunk 5]
```diff
@@ -202,3 +202,7 @@ If you have a collection of elements of a nullable type and want to filter non-n
 val nullableList: List<Int?> = listOf(1, 2, null, 4)
 val intList: List<Int> = nullableList.filterNotNull()
 ```
+
+What's next?
```
[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/null-safety.md:206]: Oops:)

File: docs/topics/jvm/java-to-kotlin-nullability-guide.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (koshachy) [reply docs/topics/jvm/java-to-kotlin-nullability-guide.md]: Oh, I think I've messed up a little bit: I also looked through another article at the same time. I guess there I found such single-line code blocks, so this comment doesn't relate to this guide. Sorry for that!

**Review by koshachy — APPROVED** [review]
LGTM!
Huge work!

**Pull request merged**